app/src/main/java/chat/rocket/android/contacts/ui/ContactsFragment.kt

```diff
@@ -314,38 +335,52 @@ class ContactsFragment : Fragment(), ContactsView {
         // Show loading while sync in progress
         recyclerView!!.visibility = View.GONE
         emptyTextView!!.visibility = View.GONE
-        showLoading()
 
         val serverUrl = serverInteractor.get()!!
         val dbManager = dbFactory.create(serverUrl)
         val contactList = dbManager.contactsDao().getAllSync()
 
-        if (contactList.isEmpty()) {
-            ui {
-                (activity as MainActivity).contactsLoadingState.observe(viewLifecycleOwner, Observer { state ->
-                    when (state) {
-                        is LoadingState.Loading -> {
-                                showLoading()
-                        }
-                        is LoadingState.Loaded -> {
+        ui {
+            (activity as MainActivity).contactsLoadingState.observe(viewLifecycleOwner, Observer { state ->
+                when (state) {
+                    is ContactsLoadingState.Loading -> {
+                        if (contactArrayList.isEmpty() && contactList.isEmpty()) {
+                            showLoading()
+                        } else {
                             hideLoading()
-                            getContactList()
-                            showToast("Contacts synced successfully", 1)
+                            if (state.fromRefreshButton) {
+                                showSpinner()
+                            }
+                            if (!loadedOnce) {
+                                getContactList()
```

Comment:
@ubarsaiyan  Just curious........  Why getContactList() here, when !loadedOnce?  When would we hit this?

Reply:
@ear-dev when we press the refresh button go somewhere else then come back the state is `Loading` and the contacts list is not empty. But we want to show the contacts if they are not loaded even once. That's when this will be hit.

Reply:
thanks